include/app/action.hpp

```diff
@@ -81,4 +90,29 @@ class Action : public QObject {
 
     QString name_;
     QString key_;
+    
+};
+
+class ActionEventFilter : public QObject
```

Comment:
I'm gonna think about this a bit more, but if you take a look at Session::Core you can see you actually have access to the actions there already, so it might be better to just modify how we store them there instead of maintaining this separate map

that way you can kind of uncouple this event filter from the action class a bit better